### PR TITLE
rec-4.1.x: Fix compilation on systems that do not define HOST_NAME_MAX

### PR DESCRIPTION
**Pull request by rgacogne**
### Short description

Backport of #9127 + #8640.

### Checklist

I have:
- [x] read the [CONTRIBUTING.md](https://github.com/PowerDNS/pdns/blob/master/CONTRIBUTING.md) document
- [x] compiled this code
- [ ] tested this code
- [ ] included documentation (including possible behaviour changes)
- [ ] documented the code
- [ ] added or modified regression test(s)
- [ ] added or modified unit test(s)
- [ ]  checked that this code was merged to master

### CODE REVIEW

File: ext/yahttp/yahttp/reqresp.cpp [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 namespace YaHTTP {
 
+  template class AsyncLoader<Request>;
```
Comment (omoerbeek) [inline ext/yahttp/yahttp/reqresp.cpp:5]: Unrelated change

Comment (rgacogne) [reply ext/yahttp/yahttp/reqresp.cpp:5]: That's the backport of #8640 mentioned in the description. I can make that into a separate backport PR but then I can't test that one on my system :)

**Comment by rgacogne** [comment]
Rebased to fix travis and spell checking failures.

File: ext/yahttp/yahttp/reqresp.cpp (hunk 1 above)
Comment (omoerbeek) [reply ext/yahttp/yahttp/reqresp.cpp:5]: OK, right.
